src/omv/modules/py_gif.c

```diff
@@ -18,114 +21,126 @@
-    py_gif_obj_t *arg_gif = gif_obj;
-    return mp_obj_new_int(arg_gif->height);
+static mp_obj_t py_gif_height(mp_obj_t self_in) {
+    py_gif_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_int(self->height);
 }
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(py_gif_height_obj, py_gif_height);
 
-static mp_obj_t py_gif_format(mp_obj_t gif_obj) {
-    py_gif_obj_t *arg_gif = gif_obj;
-    return mp_obj_new_int(arg_gif->color ? PIXFORMAT_RGB565 : PIXFORMAT_GRAYSCALE);
+static mp_obj_t py_gif_format(mp_obj_t self_in) {
+    py_gif_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_int(self->color ? PIXFORMAT_RGB565 : PIXFORMAT_GRAYSCALE);
 }
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(py_gif_format_obj, py_gif_format);
 
-static mp_obj_t py_gif_size(mp_obj_t gif_obj) {
-    #if defined(IMLIB_ENABLE_IMAGE_FILE_IO)
-    py_gif_obj_t *arg_gif = gif_obj;
-    return mp_obj_new_int(file_size(&arg_gif->fp));
-    #else
-    return mp_obj_new_int(0);
-    #endif
+static mp_obj_t py_gif_size(mp_obj_t self_in) {
+    py_gif_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_int(file_size(&self->fp));
 }
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(py_gif_size_obj, py_gif_size);
 
-static mp_obj_t py_gif_loop(mp_obj_t gif_obj) {
-    py_gif_obj_t *arg_gif = gif_obj;
-    return mp_obj_new_int(arg_gif->loop);
+static mp_obj_t py_gif_loop(mp_obj_t self_in) {
+    py_gif_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_int(self->loop);
 }
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(py_gif_loop_obj, py_gif_loop);
 
-static mp_obj_t py_gif_add_frame(uint n_args, const mp_obj_t *args, mp_map_t *kw_args) {
-    #if defined(IMLIB_ENABLE_IMAGE_FILE_IO)
-    py_gif_obj_t *arg_gif = args[0];
-    image_t *arg_img = py_image_cobj(args[1]);
-    PY_ASSERT_FALSE_MSG(IM_IS_JPEG(arg_img),   "Operation not supported on JPEG images.");
-    PY_ASSERT_FALSE_MSG(IM_IS_BINARY(arg_img), "Operation not supported on binary images.");
-    PY_ASSERT_FALSE_MSG((arg_gif->width != arg_img->w) ||
-                        (arg_gif->height != arg_img->h), "Unexpected image geometry!");
-
-    int delay = py_helper_keyword_int(n_args, args, 2, kw_args, MP_OBJ_NEW_QSTR(MP_QSTR_delay), 10);
-
-    gif_add_frame(&arg_gif->fp, arg_img, delay);
-    #else
-    mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("Image I/O is not supported"));
-    #endif
+static mp_obj_t py_gif_add_frame(uint n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_delay };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_delay, MP_ARG_INT | MP_ARG_KW_ONLY,  {.u_int = 10 } },
+    };
+
+    // Parse args.
+    py_gif_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    image_t *image = py_helper_arg_to_image(pos_args[1], ARG_IMAGE_MUTABLE);
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 2, pos_args + 2, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    // Sanity checks
+    if (self->width != image->w) {
```

Comment:
Need to check width and height.

Reply:
Check them against what ? I am checking width and height.

Reply:
Right I missed the height, thanks!